corehq/util/timezones/conversions.py

```diff
@@ -74,7 +74,9 @@ def done(self):
 
 
 def _soft_assert_tz_not_string(tz):
-    if not hasattr(tz, "localize"):
+    try:
+        assert hasattr(tz, "localize")
+    except AssertionError:
```

Comment:
It's not a good idea to rely on assertion errors for control flow. Asserts can be automatically removed by the compiler in some run modes